spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClient.java

```diff
@@ -134,12 +147,42 @@ public ServiceInstance getLocalServiceInstance() {
 
 		return instances;
 	}
-
+        
 	@Override
 	public List<String> getServices() {
-		return client.services().list()
-			.getItems()
-			.stream().map(s -> s.getMetadata().getName())
-			.collect(Collectors.toList());
+                String spelExpresion = properties.getFilter();
+		if(spelExpresion == null || spelExpresion.isEmpty()){
+                    spelExpresion = "true";
```

Comment:
I'd prefer to just have the original if there is no expression, no reason to add the overhead of spel if there is none.

```
return client.services().list()
	.getItems()
.stream().map(s -> s.getMetadata().getName())
```

Reply:
Or instead of passing a spel expression, pass the predicate, one that includes spel and one that just returns true.